test/suite/test_bug035.py

```diff
@@ -0,0 +1,83 @@
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import os
+from wtbackup import backup_base
+from wiredtiger import stat
+
+# test_backup31.py
+# This test validates a fix for a bug related to selective backup and fast truncate.
+# The bug allowed fast-truncated history store pages to reappear in the backup
+# after a shutdown. This test ensures that when a selective backup is taken,
+# only the selected tables HS data and metadata are retained, and the unwanted tables
+# are removed from HS and metadata.
+class test_bug035(backup_base):
+    conn_config = 'cache_size=1G'
+    dir='backup.dir'
+    uris = [f"table:uri_{i}" for i in range(1, 11)]
+
+    def add_timestamp_data(self, uri, key, val, timestamp):
+        self.session.begin_transaction()
+        c = self.session.open_cursor(uri, None, None)
+        for i in range(0, 1000):
+            k = key + str(i)
+            v = val
+            c[k] = v
+        c.close()
+        self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(timestamp))
+
+    def test_bug035(self):
+        for uri in self.uris:
+            # create 10 URIs and large amount of data in each.
+            self.session.create(uri, "key_format=S,value_format=S")
+            for i in range(1, 10):
+                self.add_timestamp_data(uri, "key", f"val{i}", i)
+
+        # Ensure all the data added is stable to persist data in HS.
+        self.conn.set_timestamp('stable_timestamp=' + self.timestamp_str(15))
+        self.session.checkpoint()
+
+        # Take selective backup of last first 5 tables.
+        os.mkdir(self.dir)
+        self.take_selective_backup(self.dir, [uri.replace("table:", "") + ".wt" for uri in self.uris[-5:]])
+
+        # Open the backup directory. As part of opening, it will run RTS internally to truncate any HS pages
+        # that belong to the tables that are not part of the selective backup.
+        backup_conn = self.wiredtiger_open(self.dir, "backup_restore_target=[\"{0}\"]".format('","'.join(self.uris[:5])))
```

Comment:
What is the configuration doing here?

Reply:
This config tells WT that only the files passed as part of the backup_restore_target copied were copied as part of selective backup.

Reply:
Can we put the files in a variable so that it is easier to read/understand what's happening?

Reply:
Updated in [7cfa9d6](https://github.com/wiredtiger/wiredtiger/pull/11190/commits/7cfa9d6c6aa7f36e396f930894a6cbc8915e68f2)